dvc/remote/local.py

```diff
@@ -73,6 +74,15 @@ def __init__(self, repo, config):
         else:
             self.cache_types = copy(self.DEFAULT_CACHE_TYPES)
 
+        self.hash = [utils.CHECKSUM_MD5]
+        if repo and repo.config and repo.config.config:
```

Comment:
where does it make sense to have a remote without a `repo` object?
is it possible to have a repo without config at all?

Reply:
in some tests, repo will be null, so I put this to avoid error

Reply:
what about repo without config? is a valid case? and repo.config without config?

@efiop do you recall in what cases we get remotes without a repo object?

Reply:
I don't recall anything in particular, but it is a pretty valid case.

Reply:
@efiop what about two other checks there? If we don't have any valid use cases I would prefer it to fail fast with an NPE  instead of silently applying some logic. The same goes to the `repo` object to be honest. Or we should be testing this.

Reply:
@shcheklein Indeed, the other two don't happen ever. Nice catch! We can safely replace that if with `if repo:`.